### DISCUSSION TITLE
day 1

### DISCUSSION THREAD

**Discussion by purplelemons-dev** [fuglede/adventofcode#5] [Q&A]
hello! i was working on day one and got to part 2 and ran into some trouble, but i referenced [your code](https://github.com/fuglede/adventofcode/blob/master/2023/day01/solutions.py#L16C4-L16C4) and got it to work. my original code looked like:

```py
def part2(lines:str):
    spelt = {
        "one": "1",
        "two": "2",
        ...
    }

    for k,v in spelt.items():
        lines = lines.replace(k,v)

    return part1(lines) # i ensured this handoff works correctly
```
however, when i changed the replacement to function like your code (`lines = lines.replace(k, f"{k}{v}{k}")`), it worked just fine. so, my question is: <u>why does replacing spelt-out numbers with `"one1one"` (in the case of `one`) work, but regular replacement doesn't?</u> this seems unintuitive...

**Comment by fuglede** [comment]
Try running through the logic on the string `"twone"` and see what happens.

Let me also recommend the [subreddit](https://old.reddit.com/r/adventofcode/) as a place to ask general questions; there's an army of people waiting to answer there.
